aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,295 @@
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/url-rewriting/sample)
+
+URL rewriting is the act of modifying request URLs based on one or more predefined rules. URL rewriting creates an abstraction between resource locations and their addresses so that the locations and addresses are not tightly linked. There are several scenarios where URL rewriting is valuable:
+* Moving or replacing server resources temporarily or permanently while maintaining stable locators for those resources
+* Splitting request processing across different applications or across areas of one application
+* Removing, adding, or reorganizing URL segments on incoming requests
+* Optimizing public URLs for Search Engine Optimization (SEO)
+* Permitting the use of friendly public URLs to help people predict the content they will find by following a link
+* Redirecting insecure requests to secure endpoints
+* Preventing image hotlinking
+
+You can define rules for changing the URL in several ways, including regular expression (regex) matching rules, rules based on the Apache mod_rewrite module, rules based on the IIS Rewrite Module, and with your own method and class rule logic. This document introduces URL rewriting with instructions on how to use URL Rewriting Middleware in ASP.NET Core applications.
+
+> [!NOTE]
+> URL rewriting can reduce the performance of an application. Where feasible, you should limit the number and complexity of rules.
+
+## URL redirect and URL rewrite
+The difference in wording between *URL redirect* and *URL rewrite* may seem subtle at first but has important implications for providing resources to clients. ASP.NET Core's URL Rewriting Middleware is capable of meeting the need for both.
+
+A *URL redirect* is a client-side operation, where the client is instructed to access a resource at another address. This requires a round-trip to the server, and the redirect URL returned to the client will appear in the browser's address bar when the client makes a new request for the resource. If **http://www.mysite.com/resource** is *redirected* to **http://www.mysite.com/different-resource**, the client will request **http://www.mysite.com/resource**, and the server will respond that the client should obtain the resource at **http://www.mysite.com/different-resource** with a status code indicating that the redirect is either temporary or permanent. The client will execute a new request for the resource at the redirect URL.
+
+![A WebAPI service endpoint has been temporarily changed from version 1 (v1) to version 2 (v2) on the server. A client makes a request to the service at the version 1 path /v1/api. The server sends back a 302 (Found) response with the new, temporary path for the service at version 2 /v2/api. The client makes a second request to the service at the redirect URL. The server responds with a 200 (OK) status code.](url-rewriting/_static/url_redirect.png)
```

Comment:
> It's a little odd to use an API URL example and speak of the URL being in the browser address bar.

I agree. I can revert to "URL Path" in these images instead of "Address Bar" and drop the domain. It would end up ...

URL Path: **/v1/api** to URL Path: **/v2/api**